### PR TITLE
Fix #101

### LINKED ISSUE #101: AFL++ backport: calibration failures generate stability issues
We just fixed this issue in AFL++: https://github.com/AFLplusplus/AFLplusplus/pull/425

This bug is present also upstream with a high probability (for sure IMO reading the code).

### PR DESCRIPTION
**Pull request by andreafioraldi**
Backport the fix for https://github.com/AFLplusplus/AFLplusplus/pull/425
Bug found by @dgmelski

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
Thanks!

File: afl-fuzz.c [hunk 1]
```diff
@@ -2675,6 +2675,10 @@ static u8 calibrate_case(char** argv, struct queue_entry* q, u8* use_mem,
 
 abort_calibration:
 
+  if (q->cal_failed) {
```
Comment (Dor1s) [inline afl-fuzz.c:2678]: may I ask you please to add a comment clarifying this logic?

**Comment by andreafioraldi** [comment]
Pushed an even better fix with a small note.
Basically, when calibration fails (e.g. the target is too slow or you have many parallel fuzzers and the OS is not happy) the second calibration in fuzz_one use trace_bits assuming that the current testcase was the last executed, but this is not true, it its the last execution of the previous stage.
This may leads to:
1. low stability cause the coverage of a testcase is used to measure the variability of another
2. in the worst case, the saturation of virign_bits (has_new_bits is called on the wrong map). This will saturate the fuzzer (100% bitmap usage) preventing it to discover any new coverage. A very severe bug in this case.

**Comment by andreafioraldi** [comment]
In AFL++ we are going to do a release ASAP to cope with this bug, I suggest to do the same cause it is affecting many users.
Not a problem IMO for persistent fuzzing and so not an issue for OSS-Fuzz, but users that do binary-only fuzzing with many instances are really in trouble with this bug.

**Comment by Dor1s** [comment]
Thanks @andreafioraldi, done: https://github.com/google/AFL/releases/tag/v2.57b

**Pull request merged**